### PR TITLE
add pangolin 4.3.1 and new version of pdata 1.28; minor CI update

### PR DESCRIPTION
**Pull request by kapsakcj**
Here's the diff, I upgraded to pangolin-data 1.28 & added a test for a newly designated lineage, LK.1, to the end. Everything else is the same

```
$ diff pangolin/4.3.1-pdata-1.27/Dockerfile pangolin/4.3.1-pdata-1.28/Dockerfile 
12c12
< ARG PANGOLIN_DATA_VER="v1.27"
---
> ARG PANGOLIN_DATA_VER="v1.28"
189c189,199
< column -t -s, PP218754.1-usher/lineage_report.csv
\ No newline at end of file
---
> column -t -s, PP218754.1-usher/lineage_report.csv
> 
> # new lineage LK.1 that was introduced in pango-designation v1.28: https://github.com/cov-lineages/pango-designation/commit/922795c90de355e67200cf4d379e8e5ff22472e4
> # thank you Luis, Lorraine, Marcos & team from PR Sci Trust for sharing your data!
> # genome on NCBI: https://www.ncbi.nlm.nih.gov/nuccore/2728145425
> RUN datasets download virus genome accession PP770375.1 --filename PP770375.1.zip && \
> unzip PP770375.1.zip && rm PP770375.1.zip && \
> mv -v ncbi_dataset/data/genomic.fna PP770375.1.genomic.fna && \
> rm -vr ncbi_dataset/ README.md && \
> pangolin PP770375.1.genomic.fna -o PP770375.1-usher && \
> column -t -s, PP770375.1-usher/lineage_report.csv
```

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [x] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

**Comment by kapsakcj** [comment]
`docker build` via github actions complained that `FROM` and `as` did not have the same casing, so I updated the dockerfile to have `FROM x AS app` and `FROM app AS test` to silence the warning.

I also updated the CI check to look for the uppercase `AS app` and `AS test` so that it will pass for both casings going forward.

Small/minor update to the CI

File: .github/workflows/test-PR-dockerfiles.yml [hunk 1]
```diff
@@ -60,8 +60,8 @@ jobs:
         run: |
           #checking layers
           warning=''
-          app_layer=$(grep FROM ./${{ steps.parse.outputs.tool }}/${{ steps.parse.outputs.version }}/Dockerfile | grep "as app")
-          tst_layer=$(grep FROM ./${{ steps.parse.outputs.tool }}/${{ steps.parse.outputs.version }}/Dockerfile | grep "as test")
+          app_layer=$(grep FROM ./${{ steps.parse.outputs.tool }}/${{ steps.parse.outputs.version }}/Dockerfile | grep -E "as app|AS app")
+          tst_layer=$(grep FROM ./${{ steps.parse.outputs.tool }}/${{ steps.parse.outputs.version }}/Dockerfile | grep -E "as test|AS test")
```
Comment (erinyoung) [inline .github/workflows/test-PR-dockerfiles.yml:64]: Sorry about that

**Review by erinyoung — APPROVED** [review]
Looks good! I have no changes to recommend.

**Comment by erinyoung** [comment]
I am going to 
1. merge this PR
2. deploy this to dockerhub and quay using the tag '4.3.1-pdata-1.28'
3. overwrite the 'latest' tag

**Comment by erinyoung** [comment]
You can check the status of the deploy here : https://github.com/StaPH-B/docker-builds/actions/runs/9670727959

**Comment by kapsakcj** [comment]
No worries about the CI and thanks for the quick review and deploy!

**Pull request merged**